src/acvp_util.c

```diff
@@ -289,12 +283,24 @@ int acvp_lookup_ecdsa_curve (ACVP_CIPHER cipher, char *curve_name) {
  * Convert a byte array from source to a hexadecimal string which is
  * stored in the destination.
  */
-ACVP_RESULT acvp_bin_to_hexstr (const unsigned char *src,
-                                unsigned int src_len,
-                                unsigned char *dest) {
+ACVP_RESULT acvp_bin_to_hexstr (const unsigned char *src, int src_len, char *dest, int dest_max) {
     int i, j;
     unsigned char nibb_a, nibb_b;
     unsigned char hex_chars[] = "0123456789ABCDEF";
+    
+    if (!src || !dest) {
+        return ACVP_MISSING_ARG;
+    }
+    
+    if (src_len > dest_max) {
+        return ACVP_DATA_TOO_LARGE;
+    }
+    
+    if (!src_len) {
```

Comment:
Could you remove this whole block please? The src buffer is bytes not a string, so it can contain 0x0 midway through valid data. So we need to "trust" whatever number is passed in as src_len.

Reply:
Good point, I'll remove that.